src/components/TreeList/__stories__/components/WithGroupSelectionControlledStateAndCustomIcon.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+
+import {ChevronDown, ChevronUp, Database, PlugConnection} from '@gravity-ui/icons';
+
+import {Button} from '../../../Button';
+import {Icon} from '../../../Icon';
+import {Flex, spacing} from '../../../layout';
+import {ListItemView, useListState} from '../../../useList';
+import type {KnownItemStructure} from '../../../useList';
+import {createRandomizedData} from '../../../useList/__stories__/utils/makeData';
+import {TreeList} from '../../TreeList';
+import type {TreeListOnItemClick, TreeListProps} from '../../types';
+
+/**
+ * Just for example how to work with data
+ */
+interface CustomDataStructure {
+    a: string;
+}
+
+export interface WithGroupSelectionControlledStateAndCustomIconExampleProps
+    extends Omit<
+        TreeListProps<CustomDataStructure>,
+        'value' | 'onUpdate' | 'items' | 'multiple' | 'cantainerRef' | 'size'
+    > {
+    itemsCount?: number;
+}
+
+const mapCustomDataStructureToKnownProps = (props: CustomDataStructure): KnownItemStructure => ({
+    title: props.a,
+});
+
+export const WithGroupSelectionControlledStateAndCustomIconExample = ({
+    itemsCount = 5,
+    ...props
+}: WithGroupSelectionControlledStateAndCustomIconExampleProps) => {
+    const items = React.useMemo(
+        () => createRandomizedData({num: itemsCount, getData: (a) => ({a})}),
+        [itemsCount],
+    );
+
+    const listState = useListState();
+
+    const handleItemClick: TreeListOnItemClick<CustomDataStructure> = ({id, disabled}) => {
+        if (disabled) return;
+        console.log(1);
+
+        listState.setSelected((prevState) => ({
+            [id]: !prevState[id],
+        }));
+
+        listState.setActiveItemId(id);
+    };
+
+    return (
+        <Flex direction="column" gap="3">
```

Comment:
Do we need Flex with one child?

Reply:
why not?) it's just story. I think it stayed when I tested the component with the button